kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -219,16 +222,11 @@ private void maybeStartMetricsListener(EventGroupConfig eventGroupConfig,
      * This should only be called after a successful call to {@link #startup()}.
      * @throws InterruptedException
      */
-    public void block() throws InterruptedException {
+    public void block() throws Exception {
         if (!running.get()) {
             throw new IllegalStateException("This proxy is not running");
         }
-        while (!acceptorChannels.isEmpty()) {
-            Channel channel = acceptorChannels.peek();
-            if (channel != null) {
-                channel.closeFuture().sync();
-            }
-        }
+        closeFuture.get();
```

Comment:
I was musing about suggesting `closeFuture.join()` here but checking the javadoc leads me to discover the only real difference is a checkedException from `get` vs unchecked from `join`.